CMakeLists.txt

```
@@ -354,13 +354,58 @@ if(GINKGO_BUILD_BENCHMARKS)
 endif()
 
 if(GINKGO_DEVEL_TOOLS)
-    add_custom_target(add_license
-        COMMAND ${Ginkgo_SOURCE_DIR}/dev_tools/scripts/add_license.sh
-        WORKING_DIRECTORY ${Ginkgo_SOURCE_DIR})
-    # if git-cmake-format can not build format target, do not add the dependencies
-    if(TARGET format)
+    function(enable_format_check)
+        macro(message_git_warning)
+            message(WARNING "No viable Git setup found, but GINKGO_DEVEL_TOOLS=ON is set. "
+                    "It is recommended to enable this option only with the intention of "
+                    "adding changes to Ginkgo via Git.")
+        endmacro()
+
+        find_package(Git QUIET)
+        if(NOT GIT_FOUND)
+            message_git_warning()
+            return()
+        endif()
+
+        execute_process(COMMAND ${GIT_EXECUTABLE} "rev-parse" "--git-dir"
+                        WORKING_DIRECTORY ${Ginkgo_SOURCE_DIR}
+                        RESULT_VARIABLE GINKGO_IS_GIT_DIR_ERROR)
+        if(${GINKGO_IS_GIT_DIR_ERROR})
+            message_git_warning()
+            return()
+        endif()
+
+        find_program(PRE_COMMIT pre-commit)
+        if(${PRE_COMMIT} STREQUAL "PRE_COMMIT-NOTFOUND")
```

this should also work:
```suggestion
        if(NOT PRE_COMMIT)
```